src/vs/workbench/api/common/extHostTreeViews.ts

```diff
@@ -124,7 +124,7 @@ export class ExtHostTreeViews implements ExtHostTreeViewsShape {
 			set badge(badge: vscode.ViewBadge | undefined) {
 				if (ExtHostViewBadge.isViewBadge(badge)) {
 					treeView.badge = {
-						value: Math.floor(badge.value),
+						value: Math.floor(Math.abs(badge.value)),
```

Comment:
I wonder if it's worth showing a 0 instead of absolute value. The difference between -99 and 99 is large.

Reply:
Showing a negative number here is developer error. If we just set it to zero it could hide a bug in an extension. I would be fine with showing nothing instead.